src/core/lib/iomgr/executor.c

```diff
@@ -92,6 +125,9 @@ void grpc_executor_set_threading(grpc_exec_ctx *exec_ctx, bool threading) {
   gpr_atm cur_threads = gpr_atm_no_barrier_load(&g_cur_threads);
```

Comment:
There's a lot of use of atm here, which I presume is because there may be multiple threads trying to use this at the same time (some time between executor_init and shutdown). But if that's the case, what would prevent the assert from triggering if multiple threads make it past the cur_threads check?